src/components/FileRepoSidebar.js

```diff
@@ -147,46 +123,184 @@ const FileRepoSidebar = ({ projectId, index, style, sqon, effects, ...props }) =
-                    css={`
-                      flex-grow: 1;
-                      padding-left: 15px;
-                    `}
-                    disabled={disabled}
-                    onClick={onClick}
-                  >
-                    <DownloadIcon loading={loading} />BIOSPECIMEN
-                  </Button>
-                )}
-              />
-            </div>
-          );
-        }}
-      />
+            );
+          }}
+        />
+        <ColumnsState
+          projectId={projectId}
+          graphqlField="file"
+          render={({ state }) => {
+            return (
+              <div>
+                <Heading style={{ color: '#343434', fontSize: 14, marginBottom: 5 }}>
+                  Selected File
+                </Heading>
+                <div
+                  css={`
+                    display: flex;
+                    margin-bottom: 13px;
+                  `}
+                >
+                  <LoadingOnClick
+                    onClick={async () => {
+                      downloadFile(props.selectedTableRows, gen3Key)
+                        .then(url => {
+                          let a = document.createElement('a');
+                          console.log(url);
+                          a.href = url;
+                          a.download = url.split('/').slice(-1);
+                          a.click();
+                        })
+                        .catch(err => {
+                          console.log(err);
+                          setToast({
+                            id: `${Date.now()}`,
+                            action: 'success',
+                            component: (
+                              <div
+                                css={`
+                                  display: flex;
+                                `}
+                              >
+                                <div
+                                  css={`
+                                    display: flex;
+                                    flex-direction: column;
+                                  `}
+                                >
+                                  <div
+                                    css={`
+                                      font-size: 16px;
+                                    `}
+                                  >
+                                    Failed!
+                                  </div>
+                                  <div>Unable to download file</div>
+                                  <div
+                                    css={`
+                                      color: 'red';
+                                      margin-bottom: 20px;
+                                      padding: 20px;
+                                    `}
+                                  >
+                                    <span>
+                                      Your Account doesn't have required permission to download this
```

Comment:
Your Account ~doesn't~ `does not` have `the` required permission to download this